x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/TopNOperator.java

```diff
@@ -241,11 +300,173 @@ Row row(Page origin, int rowNum, Row spare) {
+                                }
+                            }
+                            result.orderByCompositeKey.grow(result.orderByCompositeKey.length() + Long.BYTES);
+                            NumericUtils.longToSortableBytes(
+                                rowValue,
+                                result.orderByCompositeKey.bytes(),
+                                result.orderByCompositeKey.length()
+                            );
+                            result.orderByCompositeKey.setLength(result.orderByCompositeKey.length() + Long.BYTES);
+                            valueAsBytesSize = Long.BYTES;
+                        }
+                        case INT -> {
+                            int rowValue;
+                            if (mvOffset >= 0) {
+                                rowValue = result.getInt(so.channel, mvOffset);
+                            } else {
+                                rowValue = result.getInt(so.channel, 0);
+                                for (int j = 1; j < result.numberOfValues[so.channel]; j++) {
+                                    int value = result.getInt(so.channel, j);
+                                    if (sortMode == MvSortMode.MIN && value < rowValue || sortMode == MvSortMode.MAX && value > rowValue) {
+                                        rowValue = value;
+                                    }
+                                }
+                            }
+                            result.orderByCompositeKey.grow(result.orderByCompositeKey.length() + Integer.BYTES);
+                            NumericUtils.intToSortableBytes(
+                                rowValue,
+                                result.orderByCompositeKey.bytes(),
+                                result.orderByCompositeKey.length()
+                            );
+                            result.orderByCompositeKey.setLength(result.orderByCompositeKey.length() + Integer.BYTES);
+                            valueAsBytesSize = Integer.BYTES;
+                        }
+                        case DOUBLE -> {
+                            double rowValue;
+                            if (mvOffset >= 0) {
+                                rowValue = result.getDouble(so.channel, mvOffset);
+                            } else {
+                                rowValue = result.getDouble(so.channel, 0);
+                                for (int j = 1; j < result.numberOfValues[so.channel]; j++) {
+                                    double value = result.getDouble(so.channel, j);
+                                    if (sortMode == MvSortMode.MIN) {
+                                        rowValue = Math.min(value, rowValue);
+                                    } else if (sortMode == MvSortMode.MAX) {
+                                        rowValue = Math.max(value, rowValue);
+                                    }
+                                }
+                            }
```

Comment:
Why this block has a different style then the others, namely Math.min()/max()

Reply:
You're right, I'll correct it. I think I had a different approach over the past many iterations of this PR.